Common/Brokerages/KrakenBrokerageModel.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            { "BTCUSD", 0.001m },
+            { "BTCEUR", 0.001m },
+            { "BTCGBP", 0.001m },
+
+            { "BCHUSD", 0.01m },
+            { "BCHEUR", 0.01m },
+            { "BCHBTC", 0.01m },
+
+            { "ETHUSD", 0.01m },
+            { "ETHEUR", 0.01m },
+            { "ETHGBP", 0.01m },
+            { "ETHBTC", 0.01m },
+
+            { "LTCUSD", 0.1m },
+            { "LTCEUR", 0.1m },
+            { "LTCGBP", 0.1m },
+            { "LTCBTC", 0.1m }
+        };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KrakenBrokerageModel"/> class
+        /// </summary>
+        /// <param name="accountType">The type of account to be modelled, defaults to
+        /// <see cref="AccountType.Cash"/></param>
+        public KrakenBrokerageModel(AccountType accountType = AccountType.Cash)
+            : base(accountType)
+        {
+            //! Kraken support margin trading but it's disabled for now
+            if (accountType == AccountType.Margin)
+            {
+                throw new Exception("The Kraken brokerage does not currently support Margin trading.");
+            }
+        }
+
+        /// <summary>
+        /// Kraken global leverage rule
+        /// </summary>
+        /// <param name="security"></param>
+        /// <returns></returns>
+        public override decimal GetLeverage(Security security)
+        {
+            return 1m;
+        }
+
+        /// <summary>
+        /// Provides Kraken fee model
+        /// </summary>
+        /// <param name="security"></param>
+        /// <returns></returns>
+        public override IFeeModel GetFeeModel(Security security)
+        {
+            return new KrakenFeeModel();
+        }
+
+        /// <summary>
+        /// Kraken does no support update of orders
+        /// </summary>
+        /// <param name="security"></param>
+        /// <param name="order"></param>
+        /// <param name="request"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public override bool CanUpdateOrder(Security security, Order order, UpdateOrderRequest request, out BrokerageMessageEvent message)
+        {
+            message = _message;
+            return false;
+        }
+
+        /// <summary>
+        /// Evaluates whether exchange will accept order. Will reject order update
+        /// </summary>
+        /// <param name="security"></param>
+        /// <param name="order"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public override bool CanSubmitOrder(Security security, Order order, out BrokerageMessageEvent message)
+        {
+            if (order.BrokerId != null && order.BrokerId.Any())
+            {
+                message = _message;
+                return false;
+            }
+
+            decimal minimumOrderSize;
+            if (MinimumOrderSizes.TryGetValue(security.Symbol.Value, out minimumOrderSize) &&
```

Comment:
Do the lower volume coins not have minimum trade sizes? the way this is coded, if `MinimumOrderSizes` doesn't contain the pair then it assumes it can submit an order of any size.

Reply:
They do, and I need to focus more on KrakenBrokerageModel.